api/analytics/all_analytics_query.py

```diff
@@ -41,22 +43,30 @@ class Query(graphene.ObjectType):
         AllAnalytics,
         start_date=graphene.String(),
         end_date=graphene.String(),
+        location_id=graphene.Int(),
         description="Query that returns a list of all analytics")
 
-    @Auth.user_roles('Admin', 'Default User')
+    @Auth.user_roles('Admin', 'Default User', 'Super_Admin')
     def resolve_all_analytics(self, info, **kwargs):
         start_date = kwargs.get('start_date')
         end_date = kwargs.get('end_date')
+        location_id = admin_roles.user_location_for_analytics_view()
+
+        admin_details = get_user_from_db()
+        # check that id is valid
+        verify_location_id(kwargs)
+        if admin_details.roles[0].id == 3 and kwargs.get('location_id', None):
```

Comment:
Please avoid hardcoding IDs in the codebase

Reply:
@joshuaocero I'll refactor this and come up with a better implementation.